scripts/migrate_spam.py

```diff
@@ -0,0 +1,33 @@
+from website.project.model import Comment
+
+
+def migrate_comment(comment):
+    output = 'Comment fine.'
+    try:
+        temp = comment.spam_status
+    except:
+        if len(comment.reports) > 0:
+            comment.spam_status = Comment.FLAGGED
+    if comment.spam_status not in (Comment.FLAGGED, Comment.SPAM,
+                                   Comment.UNKNOWN, Comment.HAM):
+        comment.spam_status = Comment.UNKNOWN
+    try:
+        temp = comment.latest_report
+    except:
+        date = None
+        for user, report in comment.reports.iteritems():
+            report_date = report.get('date')
+            if date is None or report_date > date:
+                date = report_date
+        comment.latest_report = date
+    comment.save()
+
+
+def main():
+    comments = Comment.find()
```

Comment:
`Comment.find(Q('spam_status', 'ne', 0))` should shave some time off this script, as well as obviate some of the mess in `migrate_comment`.